domains/authorization/Authorization.API/API/Authorization/[Features]/GrantConsentToOrganizationCommand.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Data;
+using API.Models;
+using API.Repository;
+using API.ValueObjects;
+using EnergyOrigin.Domain.ValueObjects;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using OrganizationId = API.ValueObjects.OrganizationId;
+
+namespace API.Authorization._Features_;
+
+public class GrantConsentToOrganizationCommandHandler(
+    IOrganizationRepository organizationRepository,
+    IUserRepository userRepository,
+    IOrganizationConsentRepository organizationConsentRepository,
+    IUnitOfWork unitOfWork)
+    : IRequestHandler<GrantConsentToOrganizationCommand>
+{
+    public async Task Handle(GrantConsentToOrganizationCommand toOrganizationCommand, CancellationToken cancellationToken)
+    {
+        await unitOfWork.BeginTransactionAsync();
+
+        var idpUserId = IdpUserId.Create(toOrganizationCommand.UserId);
+        var organizationTin = Tin.Create(toOrganizationCommand.OrganizationCvr);
+        var consentReceiverOrganizationId = toOrganizationCommand.OrganizationId.Value;
+
+        // Throw not found if organization id is unknown
+        await organizationRepository.GetAsync(consentReceiverOrganizationId, cancellationToken);
+
+        var affiliatedOrganization = await userRepository.Query()
+            .Where(u => u.IdpUserId == idpUserId)
+            .SelectMany(u => u.Affiliations)
+            .Where(a => a.Organization.Tin == organizationTin)
+            .Select(a => a.Organization)
+            .FirstOrDefaultAsync(cancellationToken);
+
+        if (affiliatedOrganization is null)
+        {
+            throw new UserNotAffiliatedWithOrganizationCommandException();
+        }
+
+        if (affiliatedOrganization.Id == toOrganizationCommand.OrganizationId.Value)
+        {
+            throw new UnableToGrantConsentToOwnOrganization();
```

Comment:
UnableToGrantConsentToOwnOrganization <-- Maybe move this exception into exceptions folder or something, since it's used in more than one query / Command.

Reply:
Done